content/blog/duckplyr-1-1-0/index.Rmd

```diff
@@ -0,0 +1,233 @@
+---
+output: hugodown::hugo_document
+
+slug: duckplyr-1-1-0
+title: duckplyr fully joins the tidyverse!
+date: 2025-06-19
+author: Kirill Müller and Maëlle Salmon
+description: >
+    duckplyr 1.1.0 is on CRAN!
+    A drop-in replacement for dplyr, powered by DuckDB for speed.
+    It is the most dplyr-like of dplyr backends.
+
+photo:
+  url: https://www.pexels.com/photo/a-mallard-duck-on-water-6918877/
+  author: Kiril Gruev
+
+# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
+categories: [package]
+tags:
+  - duckplyr
+  - dplyr
+  - tidyverse
+---
+
+```{r include = FALSE}
+options(
+  pillar.min_title_chars = 20,
+  pillar.max_footer_lines = 7,
+  pillar.bold = TRUE
+)
+options(conflicts.policy = list(warn = FALSE))
+library(conflicted)
+conflict_prefer("filter", "dplyr", quiet = TRUE)
+```
+
+
+We're well chuffed to announce the release of [duckplyr](https://duckplyr.tidyverse.org) 1.1.0.
+This is a dplyr backend powered by [DuckDB](https://duckdb.org/), a fast in-memory analytical database system[^duckdb].
+duckplyr uses the power of DuckDB for impressive performance where it can, and seemlessly falls back to R where it can't.
+You can install it from CRAN with:
+
+[^duckdb]: If you haven't heard of it yet, watch [Hannes Mühleisen's keynote at posit::conf(2024)](https://www.youtube.com/watch?v=GELhdezYmP0&feature=youtu.be).
+
+```{r, eval = FALSE}
+install.packages("duckplyr")
+```
+
+This article shows how duckplyr can be used instead of dplyr, explain how you can help improve the package, and share a selection of further resources.
+
+## A drop-in replacement for dplyr
+
+Imagine you have to wrangle a huge dataset, like this one from the [TPC-H benchmark](https://duckdb.org/2024/04/02/duckplyr.html#benchmark-tpc-h-q1), a famous database benchmarking dataset.
+
+```{r}
+lineitem_tbl <- duckdb:::sql(
+  "INSTALL tpch; LOAD tpch; CALL dbgen(sf=1); FROM lineitem;"
+)
+lineitem_tbl <- tibble::as_tibble(lineitem_tbl)
+dplyr::glimpse(lineitem_tbl)
+```
+
+To work with this in duckplyr instead of dplyr, all you need to do is load duckplyr:
+
+```{r}
+library(duckplyr)
+```
+
+Now we can express the well-known (at least in the database community!) "TPC-H benchmark query 1" in dplyr syntax and execute it in DuckDB via duckplyr.
+
+```{r}
+tpch_dplyr <- function(lineitem) {
+  lineitem |>
+    filter(l_shipdate <= !!as.Date("1998-09-02")) |>
```

Comment:
Just curious why is the date injected? If possible it'd be better not to use `!!` in a large audience blog post as it tends to confuse users, and in this case it's especially confusing because it doesn't seem necessary? Correct me if I'm wrong, I'm not familiar with duckplyr :)